apps/studio/data/database/entity-definition-query.ts

```diff
@@ -79,32 +84,55 @@ export type EntityDefinitionVariables = {
   connectionString?: string
 }
 
-export type EntityDefinitionData = string
-export type EntityDefinitionError = ExecuteSqlError
+export async function getEntityDefinition(
+  { projectRef, connectionString, id, type }: EntityDefinitionVariables,
+  signal?: AbortSignal
+) {
+  if (!id) {
+    throw new Error('id is required')
+  }
 
-export const useEntityDefinitionQuery = <TData extends EntityDefinitionData = EntityDefinitionData>(
-  { id, type, projectRef, connectionString }: EntityDefinitionVariables,
-  { enabled = true, ...options }: UseQueryOptions<ExecuteSqlData, EntityDefinitionError, TData> = {}
-) => {
-  return useExecuteSqlQuery(
+  const sql = getEntityDefinitionSql({ id, type })
+  const { result } = await executeSql(
     {
       projectRef,
       connectionString,
-      sql: getEntityDefinitionQuery({ id, type }),
-      queryKey: databaseKeys.entityDefinition(projectRef, id),
+      sql,
+      queryKey: ['entity-definition', id],
```

Comment:
Should this be `databaseKeys.entityDefinition(projectRef, id)`?

Reply:
In this case it's just a key for the URL query string – do you think that would benefit from the whole query key and not just this part?

Reply:
yup this should be fine - executeSql doesnt cache the results in the RQ state (its not a getter since it just fires a POST request)